src/panel/event/EventListPanel.jsx

```diff
@@ -0,0 +1,132 @@
+/* global _ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import Panel from 'src/components/ui/Panel';
+import PoiEventItemList from './PoiEventItemList';
+import EventPanelError from './EventPanelError';
+import EventPanelHeader from './EventPanelHeader';
+import Telemetry from "src/libs/telemetry";
+import SearchInput from "src/ui_components/search_input";
+import nconf from '@qwant/nconf-getter';
+import IdunnPoi from "src/adapters/poi/idunn_poi";
+import events from "config/events.yml"
+
+const eventConfig = nconf.get().events;
+const MAX_PLACES = Number(eventConfig.maxPlaces);
+
+class EventListPanel extends React.Component {
+
+  static propTypes = {
+    eventName: PropTypes.string,
+    query: PropTypes.string,
+    bbox: PropTypes.string,
+  };
+
+  state = {
+    pois: [],
+    dataSource: '',
+    initialLoading: true,
+  };
+
+  componentDidMount() {
+    this.mapMoveHandler = listen('map_moveend', this.fetchData);
+
+    if (this.props.eventName) {
+      Telemetry.add(Telemetry.POI_EVENT_OPEN, null, null, { category: this.props.eventName });
+      const { label } = events.find(ev => ev.name === this.props.eventName);
+      SearchInput.setInputValue(label.charAt(0).toUpperCase() + label.slice(1));
+    }
+    this.fitMapAndFetch();
+  }
+
+  componentDidUpdate() {
+    const panelContent = document.querySelector('.panel-content');
+    if (panelContent) {
+      panelContent.scrollTop = 0;
+    }
+  }
+
+  componentWillUnmount() {
+    window.unListen(this.mapMoveHandler);
+  }
+
+  fitMapAndFetch() {
```

Comment:
I wonder if this dirty stuff, among other things, can be factorized between categories and events… I really thought events were going to be more distinct than categories, but for now it looks exactly the same :|

Reply:
Yes, I think a lot of things will be in common... basically, everything besides the list header, list items's look and the fact that we keep osm pois or not. And as you mentioned earlier, categories need a "query" param but events don't.